src/Umbraco.Core/Persistence/Repositories/Implement/EntityContainerRepository.cs

```diff
@@ -162,9 +162,12 @@ protected override void PersistDeletedItem(EntityContainer entity)
 
         protected override void PersistNewItem(EntityContainer entity)
         {
+            // TODO Ensure correct exceptions are thrown (entity.Name is not an argument and NullReferenceException shouldn't be thrown by user code)
+            if (entity == null) throw new ArgumentNullException(nameof(entity));
             EnsureContainerType(entity);
 
-            if (string.IsNullOrWhiteSpace(entity.Name)) throw new ArgumentNullOrEmptyException("entity.Name");
+            if (entity.Name == null) throw new ArgumentNullException(nameof(entity.Name));
+            if (string.IsNullOrWhiteSpace(entity.Name)) throw new ArgumentException("Value can't be empty or consist only of white-space characters.", nameof(entity.Name));
```

Comment:
This is throwing an `ArgumentException` for a value that isn't an agument (same as in `PersistUpdatedItem`): should we throw an `InvalidOperationException` instead?

Reply:
I guess the "argument" (:P) is that it's a property of an argument/parameter, but yeah i'm fine with InvalidOperationException